design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,508 @@
+VMs with Conjur by creating a Conjur host identity for their Azure resources. The Azure resource would then receive an API key as part of the 
+creation of a Conjur identity which they must use to connect to targets. This is not optimal nor convenient because we are not allowing the 
+resources to authenticate based on their Azure identity and demand that the resource hold an extra secret, the Conjur-given API key.
+
+## Solution
+
+We will add a new authenticator to Conjur, which users can authenticate with from Azure VMs. As mentioned in the [feature doc](https://app.zenhub.com/workspaces/appliance-development-5c9cf04bfa1c204b63eddcc6/issues/cyberark/conjur/1266), before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-azure/prod
+  body:
+  - !webservice
+    annotations: 
+      provider-uri: https://sts.windows.net/TENANT_ID
+      
+  - !group apps
+  
+  - !permit
+    role: !group apps
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
+
+and add the `provider-uri` annotation to the webservice (e.g `https://sts.windows.net/TENANT_ID`). 
+
+As part of our effort to improve the user experience for our authenticators, we discussed two options for defining the `provider-uri`:
+ 
+1. `provider-uri` as a variable_+
+    ```yaml
+    - !policy
+      id: conjur/authn-azure/prod
+      body:
+      - !webservice
+    
+      - !variable
+        id: provider-uri      
+    ```
+    
+1. `provider-uri` as an entry in the webservice annotations
+    ```yaml
+    - !policy
+      id: conjur/authn-azure/prod
+      body:
+        - !webservice
+          annotations: 
+            provider-uri: https://sts.windows.net/TENANT_ID
+          
+    ```     
+
+Because the `provider-uri` does not hold sensitive information, it does not need to saved in Conjur. Adding the `provider-uri` as a variable demands later that the
+admin user populate its value through the CLI. Supplying `provider-uri` as annotation entry makes it easier to set it as a key-value pair and is therefore more readable. 
+For these reasons, we have decided to add provide the `provider-uri`.
+ 
+### Azure Application Identity
+
+To authenticate with Conjur, Azure-specific fields will need to be provided in the host annotations of the Conjur host identity for the Azure resource- subscription-id, resource-group, and user/system-assigned-identity. Specifically, If the Azure resource is assigned a system-assigned-identity, 
+we will also need to validate the `oid` field of the Azure access token. These fields will be used to confirm the identity of the Azure resource that is
+sending the authentication request. This will be done by comparing the Azure access token information with the host annotations that are defined in the policy below.
```

Comment:
In this section i would add the levels of granularity that we support:
1. Subscription ID + Resource Group
2. Subscription ID + Resource Group + System Assigned Identity
3. Subscription ID + Resource Group + User Assigned Identity

We do not support a combination of System Assigned Identity & User Assigned Identity